src/components/Mermaid.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { useContext, useEffect, useMemo } from 'react'
+import MermaidControl from './MermaidControl'
+import { useDarkMode, useInkdropConfig, useMermaidRendering } from '../utils'
+
+const Mermaid: React.FC<CodeComponentProps> = ({ children }) => {
+  const id = useMemo(
+    () =>
+      `mermaid-${Math.random()
+        .toString(36)
+        .replace(/[^a-z]+/g, '')
+        .substring(0, 5)}`,
+    []
+  )
+
+  const code = useMemo(() => {
+    if (typeof children === 'string') return children
+    if (Array.isArray(children) && typeof children[0] === 'string')
+      return children[0]
+    return ''
+  }, [children])
+
+  const { theme, panZoomMode, additionalUI } = useInkdropConfig()
+
+  const { printMode } = useContext(inkdrop.markdownRenderer.Context)
+  const isDarkMode = useDarkMode()
+
+  const { error, diagramType, containerRef, parentRef, panZoomInstance } =
+    useMermaidRendering(id, code, printMode, panZoomMode, additionalUI, theme)
+
+  useEffect(() => {
+    if (!parentRef.current) return
+    const preElement = parentRef.current.parentElement
+    if (!preElement && !printMode) return
+    preElement?.setAttribute(
```

Comment:
also found a strange behavior, where the diagram dot pattern background remains in other codeblocks when opening another note:

![CleanShot 2025-02-21 at 23 24 44@2x](https://github.com/user-attachments/assets/da97fc69-a7f3-4e3c-b6ab-1e86b7e46553)

It looks like it's because `data-mermaid-theme` attribute is manually added here and cleaning up is needed.